### PR TITLE
Show move errors on the template folder management flow

### PR DESCRIPTION
**Pull request by leohemsted**
depends on https://github.com/alphagov/notifications-api/pull/2253

|||
|-|-|
| cant move to itself | image|
|cannot move subfolders|image|

### CODE REVIEW

**Comment by karlchillmaid** [comment]
@leohemsted 

`You cannot move a folder to one of its subfolders`

Does that cover all the possible scenarios?

**Comment by leohemsted** [comment]
@karlchillmaid yeah think that'll do fine 👍

File: app/main/views/templates.py [hunk 1]
```diff
@@ -125,7 +125,13 @@ def choose_template(service_id, template_type='all', template_folder_id=None):
     if request.method == 'POST' and templates_and_folders_form.validate_on_submit():
         if not can_manage_folders():
             abort(403)
-        return process_folder_management_form(templates_and_folders_form, template_folder_id)
+        try:
+            return process_folder_management_form(templates_and_folders_form, template_folder_id)
+        except Exception as e:
+            flash(e.message)
+
+    if 'templates_and_folders' in templates_and_folders_form.errors:
```
Comment (quis) [inline app/main/views/templates.py]: Is there a test for this?

Comment (leohemsted) [reply app/main/views/templates.py]: nope but there should be 👍

Comment (leohemsted) [reply app/main/views/templates.py]: and also for the exception contents from the API showing up as a flash msg

Comment (leohemsted) [reply app/main/views/templates.py]: @quis there is now a test for it

**Pull request merged**